### PR TITLE
Fix validation in Rider wizard

### PR DESCRIPTION
**Pull request by hunterwerlla**
As a follow up to #2355 , fix the issue that caused the PR to be necessary in the first place

Fix Rider wizard to do validation. There were 2 issues:
  1. It did not validate because the call to validate was not wired in
  2. After that was fixed, it still did not validate because changing the runtime/template box would not re-run validation. So, plumb another callback into the wizard to accomplish this

## Screenshot
https://user-images.githubusercontent.com/11964782/105420695-ad725b00-5bf5-11eb-9fb1-777ee491f992.mp4

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-rider/src-201-202/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt [hunk 1]
```diff
@@ -57,10 +53,13 @@ class DotNetSamProjectGenerator(
 
     // TODO: Decouple SamProjectGenerator from the framework wizards so we can re-use its panels
     private val generator = SamProjectGenerator()
-    private val samPanel = SamInitSelectionPanel(generator.wizardFragments) {
+    private val samPanel = SamInitSelectionPanel(
+        generator.wizardFragments,
+        // needed to rerun the validation when the wizard is changed
+        wizardUpdateCallback = { validateData() },
```
Comment (rli) [inline jetbrains-rider/src-201-202/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt:59]: nit: arguments are named but don't match order they were declared in constructor :(

File: jetbrains-rider/src-203+/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt [hunk 2]
```diff
@@ -56,9 +52,13 @@ class DotNetSamProjectGenerator(
 
     // TODO: Decouple SamProjectGenerator from the framework wizards so we can re-use its panels
     private val generator = SamProjectGenerator()
-    private val samPanel = SamInitSelectionPanel(generator.wizardFragments) {
-        RuntimeGroup.getById(BuiltInRuntimeGroups.Dotnet).supportedRuntimes.contains(it)
-    }
+    private val samPanel = SamInitSelectionPanel(
+        generator.wizardFragments,
+        // needed to rerun the validation when the wizard is changed
+        wizardUpdateCallback = { validateData() },
```
Comment (rli) [inline jetbrains-rider/src-203+/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt:58]: ditto

**Review by rli — APPROVED** [review]

File: jetbrains-rider/src-201-202/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-rider/src-201-202/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt]: That is funky, fixed

File: jetbrains-rider/src-203+/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt (hunk 2 above)
Comment (hunterwerlla) [reply jetbrains-rider/src-203+/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt]: fixed as well

**Pull request merged**